### ISSUE TITLE
Ludwig crashes in Google Colab

### ISSUE THREAD

**Issue by DrRaja** [ludwig-ai/ludwig#2415]
**Describe the bug**
Trying to run ludwig in Google Colab crashes the session with an "unknown" error. While trying to run the cli commands it returns NumExpr defaulting to 2 threads.
import ray failed with exception: No module named 'ray'
usage: ludwig experiment [options]
ludwig experiment: error: one of the arguments -c/--config -cs/--config_str is required

**To Reproduce**
Running the any toy example for a multi-label text classification even the ones from the official repository results in the same.

**Comment by tgaddair** [comment]
Hey @DrRaja, it looks like `ray` is not installed in your collab environment. Can you try installing Ludwig as `pip install ludwig[full]`?

Also, can you share the full stack trace? If you're not using Ray, it shouldn't need to be imported. So this could be an issue on our side.

**Comment by DrRaja** [comment]
Hey @tgaddair ,
Thank you for your response. The problem is, the Colab notebook crashes with a message "The notebook crashed with an unknown error". There's stack trace as the environment gets reset automatically. I did look into the logs and only found 
WARNING and INFO messages. The WARNING was about the protobuf version incompatibility:

[libprotobuf FATAL google/protobuf/stubs/common.cc:83] This program was compiled against version 3.9.2 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.20.1). Contact the program author for an update. If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library. (Version verification failed in "bazel-out/k8-opt/bin/tensorflow/core/framework/tensor_shape.pb.cc".)

I've tried installing both Ray and Protobuf 3.9.2 but that crashes the notebook due to incompatibility with tensorflow.
